WordPress/Jetpack/Classes/JetpackAuthenticationManager.swift

```diff
@@ -0,0 +1,72 @@
+import WordPressAuthenticator
+
+struct JetpackAuthenticationManager {
+    static func shouldPresentUsernamePasswordController(for siteInfo: WordPressComSiteInfo?, onCompletion: @escaping (WordPressAuthenticatorResult) -> Void) {
+        /// Jetpack is required. Present an error if we don't detect a valid installation.
+        guard let site = siteInfo, Self.isValidJetpack(for: site) else {
+            let viewModel = JetpackErrorViewModel(title: "Jetpack Not Found")
+            let controller = Self.errorViewController(with: viewModel)
+
+            let authenticationResult: WordPressAuthenticatorResult = .injectViewController(value: controller)
+            onCompletion(authenticationResult)
+
+            return
+        }
+
+        /// WordPress must be present.
+        guard site.isWP else {
+            let viewModel = JetpackErrorViewModel(title: "WordPress Not Installed")
+            let controller = Self.errorViewController(with: viewModel)
+
+            let authenticationResult: WordPressAuthenticatorResult = .injectViewController(value: controller)
+            onCompletion(authenticationResult)
+
+            return
+        }
+
+        /// For self-hosted sites, navigate to enter the email address associated to the wp.com account:
+        guard site.isWPCom else {
+            let authenticationResult: WordPressAuthenticatorResult = .presentEmailController
+
+            onCompletion(authenticationResult)
+
+            return
+        }
+
+        /// We should never reach this point, as WPAuthenticator won't call its delegate for this case.
+        ///
+        DDLogWarn("⚠️ Present password controller for site: \(site.url)")
+        let authenticationResult: WordPressAuthenticatorResult = .presentPasswordController(value: false)
+        onCompletion(authenticationResult)
+    }
+
+    static func presentLoginEpilogue(in navigationController: UINavigationController, for credentials: AuthenticatorCredentials, onDismiss: @escaping () -> Void) -> Bool {
+        guard Self.hasJetpackSites() == false else {
+            return false
+        }
```

Comment:
[nits] What do you think of rewriting it like this?
```suggestion
        if Self.hasJetpackSites() {
            return false
        }
```

Reply:
Yeah that makes more sense, I've been coding PHP for too long now. haha